lib/backend/etcdv3/conversion.go

```diff
@@ -91,9 +94,22 @@ func etcdToKVPair(key model.Key, ekv *mvccpb.KeyValue) (*model.KVPair, error) {
 		}
 	}
 
+	if _, ok := v.(*apiv3.GlobalNetworkPolicy); ok {
+		v.(metav1.ObjectMetaAccessor).GetObjectMeta().SetName(convertPolicyNameFromStorage(v.(metav1.ObjectMetaAccessor).GetObjectMeta().GetName()))
+	}
+
 	return &model.KVPair{
 		Key:      key,
 		Value:    v,
 		Revision: strconv.FormatInt(ekv.ModRevision, 10),
 	}, nil
 }
+
+func convertPolicyNameFromStorage(name string) string {
```

Comment:
isn't this the same function thats already defined? lets just move this into a common package to avoid duplication.

Reply:
I added these as a part of the WIP since I was testing the functionality before I started moving functions into different files and refactoring. I totally agree that these functions should be shared, but while I'm trying out a somewhat hackey solution, I'll leave them there for now for testing.